staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -196,14 +204,52 @@ func (a *GenericAdmissionWebhook) Admit(attr admission.Attributes) error {
 	hooks := hookConfig.ExternalAdmissionHooks
 	ctx := context.TODO()
 
-	errCh := make(chan error, len(hooks))
+	hookIndexCh := make(chan int, len(hooks))
 	wg := sync.WaitGroup{}
 	wg.Add(len(hooks))
 	for i := range hooks {
+		go func(hook *v1alpha1.ExternalAdmissionHook, i int) {
+			defer wg.Done()
+			if shouldCallHook(hook, attr) {
+				hookIndexCh <- i
+			}
+		}(&hooks[i], i)
+	}
+	wg.Wait()
+	close(hookIndexCh)
+
+	var hookIndice []int
```

Comment:
Do you need this at all? You can just call len() directly on the channel. If you keep it.

Reply:
Removed.